src/utils/injectFireReferrerHideableImage.js

```diff
@@ -44,6 +47,11 @@ export default ({
     if (hiddenIframe) {
       return Promise.resolve(hiddenIframe);
     }
+    const potentialHiddenFrames = selectNodes(`#${IFRAME_ATTRS.id}`);
+    if (potentialHiddenFrames.length > 0) {
+      hiddenIframe = potentialHiddenFrames[0];
+      return Promise.resolve(hiddenIframe);
+    }
     return awaitSelector(BODY).then(([body]) => {
       hiddenIframe = createNode(IFRAME, IFRAME_ATTRS, IFRAME_PROPS);
```

Comment:
I wonder if you need to move the `if (hiddenIframe)` to be inside this then. There may be multiple requests for firing an image before the `awaitSelector(BODY)` resolves. I'd like to avoid adding an ID to the page.

Reply:
> I wonder if you need to move the if (hiddenIframe) to be inside this then.

Are you suggesting something like this?

```js
return awaitSelector(BODY).then(([body]) => {
  hiddenIframe =
    hiddenIframe ?? createNode(IFRAME, IFRAME_ATTRS, IFRAME_PROPS);
  return appendNode(body, hiddenIframe);
});
```

> There may be multiple requests for firing an image before the `awaitSelector(BODY)` resolves.

That's an excellent point. I didn't think about that, but multiple requests getting to the `awaitSelector(BODY)` before the `iframe` is created makes a lot of sense.

> I'd like to avoid adding an ID to the page.

Why is that? I created an ID because I thought that selecting by ID would be a lot faster than querying by `iframe[name="Adobe Alloy"]`.

Reply:
You can see the error if you visit https://beta-sports.bwin.com/en/sports and search for the `iframe[name="Adobe Alloy"]` selector.

Reply:
The sandbox has it as well.

Reply:
I've tested this most recent version on the sandbox and it works. 
I guess I was getting a false positive race condition when I was testing #848 in the sandbox 😬